common/common.go

```diff
@@ -777,8 +782,22 @@ func DockerRunInitImage(initImage string, fName string) error {
 func untarStream(r io.Reader) error {
 
 	tr := tar.NewReader(r)
+	res := make(chan bool)
 	for {
-		header, err := tr.Next()
+
+		var header *tar.Header
+		var err error
+
+		go func() {
+			header, err = tr.Next()
+			res <- true
```

Comment:
go-routine leak, if timer below fires then there's no consumer on this chan.

Reply:
The original problem was that the `tr.Next()` blocks forever, and prevents the program from terminating.

Now, the timeout will fire, return an error, and the CLI will terminate the program (see above manual testing).

I'm attempting to follow this pattern: https://gobyexample.com/timeouts

Perhaps unsuccessfully? Could you suggest an alternative?

Reply:
Nvm. I think I've figured out a more elegant approach.